templates/ovncontroller/bin/readiness.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/sh
+
+# Function to handle errors and exit with a message
+error_exit() {
+    echo "$1"
+    exit 1
+}
+
+# Check if ovn-controller is connected to the OVN SB database
+check_ovn_controller_connection() {
+    output=$(ovn-appctl -t ovn-controller connection-status)
+    if [ $? -ne 0 ]; then
+        error_exit "ERROR - Failed to get connection status from ovn-controller"
+    fi
+    if [ "$output" != "connected" ]; then
+        error_exit "ERROR - OVN-Controller is not connected to OVN SB database"
+    fi
+}
+
+# Check if ovsdb-server is running and list its databases
+check_ovsdb_server() {
```

Comment:
Question:

Shouldn't this check belong to where the ovsdb-server actually runs - to the `-ovs` pod checks? From `ovn-controller` pod perspective, you already confirmed that `connection-status` is fine for SB. I don't think there's an interface to check that its vswitchd openflow socket and ovsdb-server local connection is established. Maybe OVN should add commands for these?

Reply:
Yes, I understand your comment that the last two checks for ovsdb-server and ovs-vswitchd are directly related to the ready status of those services. For ovn-controller, it should be sufficient to confirm that the connection-status = 'connected', indicating it's connected to the database.

Considering discussions with other stakeholders, such as the telemetry team, who are concerned about the time impact of running health checks every second, we should aim to keep these checks as simple as possible. Each service should only run the necessary checks, and we should trust that other services will do the same. Everyone is responsible for ensuring their own checks run in a timely manner.

So removing last checks over ovsdb-server and ovs-vswitchd.

Reply:
Have you removed the checks? I still see them @fernandoroyosanchez

Reply:
It makes sense now that I understood the ticket referred to creating liveness/readiness probes for all the containers and not just for ovn-controller. Thanks for insisting!